tabular/src/autogluon/tabular/models/rf/rf_model.py

```diff
@@ -358,3 +477,39 @@ def _more_tags(self):
         else:
             tags['valid_oof'] = True
         return tags
+
+    def _valid_compilers(self):
+        return [RFNativeCompiler, RFOnnxCompiler]
+
+    def _get_compiler(self):
+        compiler = self.params_aux.get('compiler', None)
+        compilers = self._valid_compilers()
+        compiler_names = {c.name: c for c in compilers}
+        if compiler is not None and compiler not in compiler_names:
+            raise AssertionError(f'Unknown compiler: {compiler}. Valid compilers: {compiler_names}')
+        if compiler is None:
+            return RFNativeCompiler
+        compiler_cls = compiler_names[compiler]
+        if not compiler_cls.can_compile():
+            compiler_cls = RFNativeCompiler
```

Comment:
Add logic here so we can test to ensure that the compiler actually works in unit tests:

```
compiler_fallback = self.params_aux.get('compiler_fallback_to_native', True)
if not compiler_cls.can_compile():
    if not compiler_fallback:
         raise AssertionError(f'Specified compiler ({compiler}) is unable to compile (potentially lacking dependencies) and "compiler_fallback==False"')
    compiler_cls = RFNativeCompiler
```

Then change unit tests to specify `compiler_fallback_to_native=False`